nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBReference.java

```diff
@@ -126,7 +129,8 @@ public void addLinks(List<String> sourceIds, String targetId, Session session) {
         MongoCollection<Document> coll = getCollection(mongodbSession);
         List<Document> newDocs = sourceIds.stream()
                                           .map(sourceId -> buildDoc(sourceId, targetId))
-                                          .filter(doc -> coll.count(doc) == 0)
+                                          .filter(doc -> countDocuments(mongodbSession.getDirectory().databaseID, coll,
```

Comment:
Please move `mongodbSession.getDirectory().databaseID` to a local variable to have it be evaluated just once.